### PR TITLE
fix(cppcheck_CI): fix unknownMacro

### PR DESCRIPTION
**Pull request by kobayu858**
## Description
This is a fix based on cppcheck unknownMacro warnings.
The problem is that files with this error are not parsed well.
This PR solves that problem at the same time.

```
common/bag_time_manager_rviz_plugin/src/bag_time_manager_panel.hpp:44:11: error: There is an unknown macro here somewhere. Configuration is required. If Q_SLOTS is a macro then please configure it. [unknownMacro]
protected Q_SLOTS:
          ^
common/polar_grid/src/polar_grid_display.hpp:81:9: error: There is an unknown macro here somewhere. Configuration is required. If Q_SLOTS is a macro then please configure it. [unknownMacro]
private Q_SLOTS:
        ^
common/tier4_adapi_rviz_plugin/src/door_panel.hpp:58:9: error: There is an unknown macro here somewhere. Configuration is required. If slots is a macro then please configure it. [unknownMacro]
private slots:
        ^
common/tier4_adapi_rviz_plugin/src/route_panel.hpp:72:9: error: There is an unknown macro here somewhere. Configuration is required. If slots is a macro then please configure it. [unknownMacro]
private slots:
        ^
common/tier4_adapi_rviz_plugin/src/state_panel.hpp:67:8: error: There is an unknown macro here somewhere. Configuration is required. If Q_SLOTS is a macro then please configure it. [unknownMacro]
public Q_SLOTS:  // NOLINT for Qt
       ^
common/tier4_camera_view_rviz_plugin/src/bird_eye_view_tool.hpp:88:9: error: There is an unknown macro here somewhere. Configuration is required. If Q_SLOTS is a macro then please configure it. [unknownMacro]
private Q_SLOTS:
        ^
common/tier4_camera_view_rviz_plugin/src/third_person_view_tool.hpp:88:9: error: There is an unknown macro here somewhere. Configuration is required. If Q_SLOTS is a macro then please configure it. [unknownMacro]
private Q_SLOTS:
        ^
common/tier4_localization_rviz_plugin/src/pose_history_footprint/display.hpp:63:9: error: There is an unknown macro here somewhere. Configuration is required. If Q_SLOTS is a macro then please configure it. [unknownMacro]
private Q_SLOTS:
        ^
common/tier4_localization_rviz_plugin/src/pose_with_covariance_history/pose_with_covariance_history_display.hpp:65:9: error: There is an unknown macro here somewhere. Configuration is required. If Q_SLOTS is a macro then please configure it. [unknownMacro]
private Q_SLOTS:
        ^
common/tier4_planning_rviz_plugin/src/path/display.cpp:138:1: error: There is an unknown macro here somewhere. Configuration is required. If PLUGINLIB_EXPORT_CLASS is a macro then please configure it. [unknownMacro]
PLUGINLIB_EXPORT_CLASS(rviz_plugins::AutowarePathWithLaneIdDisplay, rviz_common::Display)
^
common/tier4_planning_rviz_plugin/src/tools/max_velocity.hpp:61:9: error: There is an unknown macro here somewhere. Configuration is required. If Q_SLOTS is a macro then please configure it. [unknownMacro]
private Q_SLOTS:
        ^
common/tier4_state_rviz_plugin/src/include/autoware_state_panel.hpp:88:8: error: There is an unknown macro here somewhere. Configuration is required. If Q_SLOTS is a macro then please configure it. [unknownMacro]
public Q_SLOTS:  // NOLINT for Qt
       ^
common/tier4_system_rviz_plugin/src/mrm_summary_overlay_display.hpp:82:9: error: There is an unknown macro here somewhere. Configuration is required. If Q_SLOTS is a macro then please configure it. [unknownMacro]
private Q_SLOTS:
        ^
common/tier4_traffic_light_rviz_plugin/src/traffic_light_publish_panel.hpp:50:8: error: There is an unknown macro here somewhere. Configuration is required. If Q_SLOTS is a macro then please configure it. [unknownMacro]
public Q_SLOTS:
       ^
common/tier4_vehicle_rviz_plugin/src/tools/acceleration_meter.hpp:48:9: error: There is an unknown macro here somewhere. Configuration is required. If Q_SLOTS is a macro then please configure it. [unknownMacro]
private Q_SLOTS:
        ^
common/tier4_vehicle_rviz_plugin/src/tools/console_meter.hpp:49:9: error: There is an unknown macro here somewhere. Configuration is required. If Q_SLOTS is a macro then please configure it. [unknownMacro]
private Q_SLOTS:
        ^
common/tier4_vehicle_rviz_plugin/src/tools/steering_angle.hpp:47:9: error: There is an unknown macro here somewhere. Configuration is required. If Q_SLOTS is a macro then please configure it. [unknownMacro]
private Q_SLOTS:
        ^
common/tier4_vehicle_rviz_plugin/src/tools/turn_signal.hpp:46:9: error: There is an unknown macro here somewhere. Configuration is required. If Q_SLOTS is a macro then please configure it. [unknownMacro]
private Q_SLOTS:
        ^
common/tier4_vehicle_rviz_plugin/src/tools/velocity_history.hpp:50:9: error: There is an unknown macro here somewhere. Configuration is required. If Q_SLOTS is a macro then please configure it. [unknownMacro]
private Q_SLOTS:
        ^
planning/behavior_path_planner/autoware_behavior_path_external_request_lane_change_module/src/manager.cpp:45:1: error: There is an unknown macro here somewhere. Configuration is required. If PLUGINLIB_EXPORT_CLASS is a macro then please configure it. [unknownMacro]
PLUGINLIB_EXPORT_CLASS(
^
planning/behavior_path_planner/autoware_behavior_path_lane_change_module/src/manager.cpp:429:1: error: There is an unknown macro here somewhere. Configuration is required. If PLUGINLIB_EXPORT_CLASS is a macro then please configure it. [unknownMacro]
PLUGINLIB_EXPORT_CLASS(
^
planning/behavior_velocity_planner/autoware_behavior_velocity_intersection_module/src/manager.cpp:594:1: error: There is an unknown macro here somewhere. Configuration is required. If PLUGINLIB_EXPORT_CLASS is a macro then please configure it. [unknownMacro]
PLUGINLIB_EXPORT_CLASS(
^
simulator/tier4_dummy_object_rviz_plugin/src/tools/interactive_object.hpp:142:11: error: There is an unknown macro here somewhere. Configuration is required. If Q_SLOTS is a macro then please configure it. [unknownMacro]
protected Q_SLOTS:
          ^
simulator/tier4_dummy_object_rviz_plugin/src/tools/delete_all_objects.hpp:75:9: error: There is an unknown macro here somewhere. Configuration is required. If Q_SLOTS is a macro then please configure it. [unknownMacro]
private Q_SLOTS:
        ^
```
## Related links

- https://files.cppchecksolutions.com/manual.pdf

**Parent Issue:**

## Notes for reviewers

None.

## Interface changes

None.

## Effects on system behavior

None.

### CODE REVIEW

File: cppcheck_macros.hpp [hunk 1]
```diff
@@ -0,0 +1,25 @@
+// Copyright 2024 Tier IV, Inc.
```
Comment (veqcc) [inline cppcheck_macros.hpp:1]:
```suggestion
// Copyright 2024 TIER IV, Inc.```

File: cppcheck_macros.hpp [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef CPPCHECK_MACROS_HPP_
+#define CPPCHECK_MACROS_HPP_
+
+// Empty macro definition (so that cppcheck ignores the macro)
```
Comment (veqcc) [inline cppcheck_macros.hpp:18]: > (so that cppcheck ignores the macro)

This comment is redundant. You can remove it

Comment (kobayu858) [reply cppcheck_macros.hpp]: Thank you for the advice. I have removed the redundant comment.

**Comment by kobayu858** [comment]
@veqcc 

By using the `-D` option, we were able to make the macro understood without creating a new file.
I have checked locally, but have not checked the CI because I was not sure how to check it.

File: cppcheck_macros.hpp (hunk 1 above)
Comment (kobayu858) [reply cppcheck_macros.hpp]: This file has been deleted.

**Comment by kobayu858** [comment]
@veqcc 

The file `signal_display.cpp` in which `PLUGINLIB_EXPORT_CLASS` exists has been modified and tested for CI.

If the command is wrong, I get the following error.

> common/autoware_overlay_rviz_plugin/autoware_overlay_rviz_plugin/src/signal_display.cpp:499:1: error: syntax error [syntaxError]
> PLUGINLIB_EXPORT_CLASS(autoware_overlay_rviz_plugin::SignalDisplay, rviz_common::Display)
> ^

However, since it did not appear this time, I was able to confirm that it is working without any problem.

**Review by veqcc — APPROVED** [review]

**Pull request merged**